java/src/jmri/jmrix/dccpp/swing/DCCppTableModel.java

```diff
@@ -98,11 +98,11 @@ public boolean contains(List<Object> v) {
     }
 
     public void insertData(List<Object> v, boolean isnew) {
+        v.add(Bundle.getMessage("ColumnDelete"));
+        v.add(isnew); // is new
+        v.add(false); // is dirty (no)
+        v.add(false); // is marked for delete (of course not)
         if (!rowData.contains(v)) {
-            v.add(Bundle.getMessage("ColumnDelete"));
-            v.add(isnew); // is new
-            v.add(false); // is dirty (no)
-            v.add(false); // is marked for delete (of course not)
             rowData.add(v);
```

Comment:
Why is things added to `v` if `v` is not added to `rowData`?

Reply:
So that the if works. Without this change, the !contains was always true.

Reply:
My apologize. I forgot that v is a list and that equality of two lists is if they have the same elements or not.